simln-lib/src/test_utils.rs

```diff
@@ -45,3 +53,140 @@ pub fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
         })
         .collect()
 }
+
+// Mock implementations for tests
+#[cfg(test)]
+mock! {
+    pub Generator {}
+
+    impl fmt::Display for Generator {
+        fn fmt<'a>(&self, f: &mut fmt::Formatter<'a>) -> fmt::Result;
+    }
+
+    impl PaymentGenerator for Generator {
+        fn payment_start(&self) -> Option<Duration>;
+        fn payment_count(&self) -> Option<u64>;
+        fn next_payment_wait(&self) -> Result<Duration, PaymentGenerationError>;
+        fn payment_amount(&self, destination_capacity: Option<u64>) -> Result<u64, PaymentGenerationError>;
+    }
+}
+
+#[cfg(test)]
+mock! {
+    pub LightningNode {}
+    #[async_trait]
+    impl crate::LightningNode for LightningNode {
+        fn get_info(&self) -> &NodeInfo;
+        async fn get_network(&mut self) -> Result<bitcoin::Network, LightningError>;
+        async fn send_payment(
+                &mut self,
+                dest: bitcoin::secp256k1::PublicKey,
+                amount_msat: u64,
+            ) -> Result<lightning::ln::PaymentHash, LightningError>;
+        async fn track_payment(
+                &mut self,
+                hash: &lightning::ln::PaymentHash,
+                shutdown: triggered::Listener,
+            ) -> Result<crate::PaymentResult, LightningError>;
+        async fn get_node_info(&mut self, node_id: &PublicKey) -> Result<NodeInfo, LightningError>;
+        async fn list_channels(&mut self) -> Result<Vec<u64>, LightningError>;
+    }
+}
+
+/// Type alias for the result of setup_test_nodes
+type TestNodesResult = (
+    Vec<NodeInfo>,
+    HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
+);
+
+/// Creates a set of nodes with mock implementations
+#[cfg(test)]
+pub fn setup_test_nodes(node_count: usize, keysend_indices: &[usize]) -> TestNodesResult {
```

Comment:
This is pretty fussy, so feel free to ignore.

I'm often peeved by test helper functions that have parameters for a very specific thing because they're inflexible - you often end up creating multiple helpers or adding tons of parameters that aren't necessary. I feel like we're hitting this with `setup_test_nodes` and `setup_network_test_nodes`?

Perhaps we could use functional options or a [builder](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html) pattern here to have one single network creation util that's flexible enough to serve different use cases?